### ISSUE TITLE
pacman yes/no prompt broken

### ISSUE THREAD

**Issue by Craie** [msys2/MSYS2-packages#3532]
### Description / Steps to reproduce the issue

1. run pacman and when you get to a prompt such as Y/n, enter capital Y instead of just hitting Enter

### Expected behavior

Either y or Y or Enter should go forth with the procedure on such a prompt.

(You might check the y/N prompts for similar issues.)

### Actual behavior

The program will end/skip that step.  This happens particularly for installing new devel packages the first time with a new system with this command:

`    pacman -S --needed base-devel mingw-w64-x86_64-toolchain`

When it asks to proceed and my students don't hit Enter but type a Y first (they are first semester CSC -- forgive them), it quits instead of running.

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

I don't have my Windows computer with me now.  It is Win10, though.  I will edit this once I get to it.

**Comment by Biswa96** [comment]
The command `pacman -S --needed base-devel mingw-w64-x86_64-toolchain` is installing a group package. So, it will first ask with this prompt

```
Enter a selection (default=all):
```

Just press <kbd>Enter</kbd> at this step.

Then pacman will ask if user want to install the packages or not with this prompt

```
:: Proceed with installation? [Y/n] 
```

Now press <kbd>Y</kbd> at this step.

**Comment by Craie** [comment]
That's what I did and my students did on a fresh install and Y did not go forth but dropped out of the program.  Enter does, but capital Y does not.  (I did not yet try little y.)

**Comment by Biswa96** [comment]
After pressing <kbd>Y</kbd>, one has to press <kbd>Enter</kbd> also.

If prompts did not work, one can skip those with `--noconfirm` option. So, the final command would be like 

```
pacman -S --needed --noconfirm base-devel mingw-w64-x86_64-toolchain
```

Double check the command before running it. The `--noconfirm` option will not stop until the installation procedure is completed.

**Comment by Craie** [comment]
Yes, we pressed Enter after the Y.  That's when the program ends.  There is clearly a missing tolower or the like in the code.

Students already find it hard enough to type the original command.  Adding another parameter will hardly help that.  But it is a valid suggestion so thank you.

I still think someone should fix the issue since it is such a minor fix and such a tedious error for those new to such products.

**Comment by Craie** [comment]
I know what the Y/n means.  I told my students to just hit Enter.  But they are first semester and don't always pay attention to details.  But they know to be fairly literal already and so they are hitting a capital Y at this prompt.

I've tried it out myself and doing a capital Y (with Enter) doesn't work.  It just stops the program.  Maybe someone fixed it in the last few days.  It has been since Tuesday since I've done one of these installs.  But it wasn't working then.

Even if you say it "does not mean it works with uppercase Y", it should from a pure UI and usability perspective.

I'm trying to be a good citizen of the open source world here and let you know of a problem.

**Comment by Craie** [comment]
Okay.  I finally got to my Windows 10 box and tried a clean install.  It is working fine there.  Apparently the students' whose boxes weren't working were Windows 11.  They barely look different.  I'm sorry I got that wrong.  I don't have Win 11 to work with as my wife is out of town now.

**Comment by sskras** [comment]
@Craie commented [15 hours ago](https://github.com/msys2/MSYS2-packages/issues/3532#issuecomment-1416437734):

> Apparently the students' whose boxes weren't working were Windows 11. They barely look different.

Could it be that the physical keyboard layout was QWERTY, but the virtual layout / the language was German (QWERTZ) ?

**Comment by Craie** [comment]
It was several students and we aren't in a heavily German area (Chicago suburbs).  Seems doubtful.  I'll try to talk to the students in question next week and get more details.

**Comment by lazka** [comment]
I've had pacman drop the input for those prompts in random places as you described. I never managed to reliably reproduce it though.

**Comment by lazka** [comment]
One case is if you type "y" before it asks you, then if you type "y" at the question again it will quit.

**Comment by lazka** [comment]
I've opened https://github.com/msys2/msys2-runtime/issues/199 for the underlying cause

**Comment by lazka** [comment]
> I've opened [msys2/msys2-runtime#199](https://github.com/msys2/msys2-runtime/issues/199) for the underlying cause

This is fixed now with #5513

In theory things could be changed to repeat the prompt in case an invalid answer is given (similar to when installing a group). If someone thinks that is an improvement I could ask upstream pacman what they think about it.
